src/lib/ares_process.c

```diff
@@ -147,6 +155,16 @@ static void server_set_good(ares_server_t *server, ares_bool_t used_tcp)
 
   if (server->consec_failures > 0) {
     server->consec_failures = 0;
+    reinsert                = ARES_TRUE;
+  }
+
+  server->consec_successes++;
+  if (server->consec_successes >= channel->min_consec_successes) {
```

Comment:
Should this also check if `server->is_failed` is false?  Currently, if a server is up and working, you'll be removing it from the list and reinserting it after every request (after the first `channel->min_consec_successes`) despite its position not changing.